server/chat-plugins/quotes.ts

```diff
@@ -37,13 +37,29 @@ export const commands: ChatCommands = {
 		room = this.requireRoom();
 		const roomQuotes = quotes[room.roomid];
 		if (!roomQuotes?.length) return this.errorReply(`This room has no quotes.`);
-		this.runBroadcast(true);
 		const {quote, date, userid} = roomQuotes[Math.floor(Math.random() * roomQuotes.length)];
 		const time = Chat.toTimestamp(new Date(date), {human: true});
 		const attribution = toID(target) === 'showauthor' ? `<hr /><small>Added by ${userid} on ${time}</small>` : '';
 		return this.sendReplyBox(`${Chat.getReadmoreBlock(quote)}${attribution}`);
 	},
 	randquotehelp: [`/randquote [showauthor] - Show a random quote from the room. Add 'showauthor' to see who added it and when.`],
+	
+	nquote(target, room, user) {
+		room = this.requireRoom();
+		const roomQuotes = quotes[room.roomid];
+		if (!roomQuotes) return this.errorReply(`No quotes found for ${room.roomid}.`);
+		const index = parseInt(target.trim());
+		if (isNaN(index)) {
+			return this.errorReply(`Invalid index.`);
+		}
+		if (!roomQuotes[index - 1]) {
+			return this.errorReply(`Out of bounds.`);
+		}
```

Comment:
```suggestion
		if (isNaN(index) || !roomQuotes[index - 1]) {
			return this.errorReply(`Invalid index.`);
		}
```
Logical Operators like `||` will short circuit if needed. This means that if the whole condition is 100% going to be true or false, the rest of the condition will not be checked. In this case if `index` is NaN, it won't attempt to check if there is a quote at `index - 1`. Only if index is not NaN will that be checked.